config/src/main/java/com/alibaba/nacos/config/server/controller/HistoryController.java

```diff
@@ -86,7 +94,11 @@ public ConfigHistoryInfo getConfigHistoryInfo(@RequestParam("nid") Long nid) {
      * @since 1.4.0
      */
     @GetMapping(value = "/previous")
-    public ConfigHistoryInfo getPreviousConfigHistoryInfo(@RequestParam("id") Long id) {
+    @Secured(action = ActionTypes.READ, parser = ConfigResourceParser.class)
+    public ConfigHistoryInfo getPreviousConfigHistoryInfo(@RequestParam("dataId") String dataId, // for @Secured
```

Comment:
Please do not change the paramters

Reply:
persistService find detailConfigHistory add params dataId, group, tenant